### PR TITLE
Update

### PR DESCRIPTION
**Pull request by jholbrook64**
Merging to own forked repo,  first merge to main on forked repo. Added changes to allow for grouping of large dataset.

### CODE REVIEW

File: src/time_summaries.R [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  
+  # returns dataframe
+  return(annual)
+}
+
+evaluate_monthly <- function(in_file)
+{
+  dat <- readRDS(in_file)
+  monthly <- dat %>% 
+    mutate(month_year = lubridate::floor_date(date, "month")) %>% #lapply(date, (format(as.Date(date), "%Y-%m")))
```
Comment (limnoliver) [inline src/time_summaries.R:43]: I think this line works the same as `lubridate::ym(date)` though maybe you're wanting to retain an actual date.

File: src/time_summaries.R [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    group_by(seg_id_nat, month_year) %>% 
+    drop_na(seg_id_nat) %>% 
+    summarize(n_per_month = length(unique(date))) %>%
+    group_by(seg_id_nat)
```
Comment (limnoliver) [inline src/time_summaries.R]: Do you mean `ungroup` here?

File: src/make_plots.R [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  #convert((num(n_per_month)))
+  print(in_monthDat)
+  
+  in_monthDat <- readr::read_csv(in_monthDat)
+  #in_monthDat <- as.numeric(in_monthDat$n_per_month)
+  
+  months <- in_monthDat %>% 
+    # uses simple val 27 in absence of costly date check function
+    filter(n_per_month > 27) %>% 
+    subset(month_year < "1995-01-01") %>%
```
Comment (limnoliver) [inline src/make_plots.R]: Should this be another filter? Not sure why you chose `subset` over `filter`. 

Also, why are you filtering to dates less than 1995? Seems like if you want to count the number of observations in all time (as you do below on line 27), you'd want to keep all data.  Did you mean dates > 1995? I think I would be in favor of doing the initial analysis on all data, and we can use this 1995 criteria later?

File: src/make_plots.R [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    # uses simple val 27 in absence of costly date check function
+    filter(n_per_month > 27) %>% 
+    subset(month_year < "1995-01-01") %>% 
+    group_by(seg_id_nat) %>% 
+    mutate(n_all_time= sum(n_per_month)) #%>% 
+   # mutate(months = count(month_year))
+  
+  v <- 0 
+  for (i in 1:nrow(months)) {  # nots ure why this is computing out the way it is. 
+    if (months[i, 4] > 10000)  #this is the check the amount of data avaliable.
```
Comment (limnoliver) [inline src/make_plots.R]: I think you're trying to find the number of sites with >10,000 obs, but I'm seeing a few issues with this calculation.

First, you're only counting complete months (maybe that's what you want), but if you want to a site to have a lot of January observations (for example), this check won't ensure that you have enough complete Januaries to do a trend analysis. Second, above, you're grouping and mutating, which means you're maintaining all rows in your `n_all_time` calculation. This means if a site does have >10,000 observations, you'll count in `n_all_time` times, which I suspect is why it's not counting the way you expect. 

Instead, if you did a `group_by` and `summarize` to calculate `n_all_time`, then you could do `sum(months$n_all_time > 10000)` to make the same calculation.

**Review by limnoliver — CHANGES_REQUESTED** [review]
Hey Jack - I've included a few comments on your monthly calculation code, but did not review the rest.

After you've addressed my comments, it would be helpful in the conversation tab to outline what the goal of this current analysis is, and maybe screen shot or paste some of the output of your target (e.g., how many sites have 20 years worth of complete July data to evaluate, or whatever your target is).

File: src/make_plots.R (hunk 4 above)
Comment (jholbrook64) [reply src/make_plots.R]: I will try this!

**Comment by jholbrook64** [comment]
> 
> 
> Hey Jack - I've included a few comments on your monthly calculation code, but did not review the rest.
> 
> After you've addressed my comments, it would be helpful in the conversation tab to outline what the goal of this current analysis is, and maybe screen shot or paste some of the output of your target (e.g., how many sites have 20 years worth of complete July data to evaluate, or whatever your target is).

sounds good - I will post an outline on the share point!

File: src/time_summaries.R (hunk 2 above)
Comment (jholbrook64) [reply src/time_summaries.R]: fixed

File: src/time_summaries.R (hunk 1 above)
Comment (jholbrook64) [reply src/time_summaries.R:43]: I'm not sure if lubridate::ym() is a real function, it would make sense they include it, but they don't seem to.

**Comment by jholbrook64** [comment]
My updated goal for the [trend analysis](https://doimspp.sharepoint.com/:w:/r/sites/IIDDStaff/_layouts/15/Doc.aspx?sourcedoc=%7B1423E717-8B9A-4116-8F82-0694F725DA9B%7D&file=Methods%20for%20Site%20selection.docx&action=default&mobileredirect=true)

File: src/make_plots.R (hunk 4 above)
Comment (jholbrook64) [reply src/make_plots.R]: fixed this

File: src/make_plots.R (hunk 3 above)
Comment (jholbrook64) [reply src/make_plots.R]: fixed, has similar solution that uses filter

**Pull request closed without merging**